checkout/src/main/java/com/checkout/validation/validator/CvvValidator.kt

```diff
@@ -41,9 +41,18 @@ internal class CvvValidator : Validator<CvvValidationRequest, Unit> {
                 ValidationError.CVV_CONTAINS_NON_DIGITS,
                 "CVV should contain only digits"
             )
+            cvvLength.contains(0) && cvv.isNotEmpty() && cvvLength.any { cvv.length < it } -> throw ValidationError(
+                ValidationError.CVV_INCOMPLETE_LENGTH,
+                "Incomplete CVV length, it should be $cvvLength"
+            )
+            !cvvLength.contains(0) && cvvLength.any { cvv.length < it } && !cvvLength.contains(cvv.length)
+            -> throw ValidationError(
+                ValidationError.CVV_INCOMPLETE_LENGTH,
+                "Incomplete CVV length, it should be $cvvLength"
+            )
```

Comment:
We have first condition which throws the error when cvv is empty.
We have second condition which throws the error  when cvv contains non-digits.
And we have the last condition which throws the error when card scheme doesn't support such a length.

We have all we need on the plate, just one condition needed. It can be optimized, just shuffle it a bit.
F.e., off top of my head, just try to optimize as you see it:

    private fun validateCvv(cvv: String, cvvLength: Set<Int>) {
        when {
            cvv.any { !it.isDigit() } -> throw ValidationError(
                ValidationError.CVV_CONTAINS_NON_DIGITS,
                "CVV should contain only digits"
            )

            cvvLength.any { it == cvv.length} -> return

            cvv.length < cvvLength.min() -> throw ValidationError(
                ValidationError.CVV_INCOMPLETE_LENGTH,
                "Incomplete CVV length, it should be $cvvLength"
            )

            else -> throw ValidationError(
                ValidationError.CVV_INVALID_LENGTH,
                "CVV length should be $cvvLength"
            )
        }
    }

Reply:
Yes , it can be improved better in this way. However,
1. For maestro, the length is 0 or 3 for CVV. So, the condition is improved as min and max as below:

cvv.length < cvvLength.min() || cvv.length < cvvLength.max() -> throw ValidationError(
                ValidationError.CVV_INCOMPLETE_LENGTH,
                "Incomplete CVV length, it should be $cvvLength"
            )
            
2. for card scheme, if the cvvlength is invalid or not identified. We are throwing an invalid card scheme for that cvv length which was already in the file. Do we still dont required this?

Reply:
1. You're right it's the tricky part, mb users want to go without CVV and entered one symbol by mistake or want to use 3-digit CVV. So it's up to you.
2. Yep, forgot to add it in a sample.

Reply:
Btw we have empty set for UNKNOWN.

Reply:
Improved as discussed